src/components/BigNumber/BigNumber.tsx

```diff
@@ -0,0 +1,273 @@
+		icon,
+		orientation,
+		height,
+		chartWidth,
+		lineProps
+	);
+	return (
+		<div tabIndex={0} className="big-number-container">
+			<Grid
+				width={chartWidth}
+				height={height}
+				areas={areas}
+				columns={columns}
+				columnGap="size-50"
+				justifyItems={generalJustify}
+				alignItems={'center'}
+			>
+				{lineProps && (
+					<View gridArea="sparkline" justifySelf={'center'}>
+						<RscChart chartWidth={cWidth} height={cHeight} data={data} locale={locale} {...rscChartRemain}>
+							<Line {...lineProps} isSparkline isMethodLast={method === 'last'} padding={padding} />
+						</RscChart>
+					</View>
+				)}
+				{displayCombo && icon ? (
+					<View gridArea="combo" alignSelf={labelAlign} UNSAFE_className="big-number-label">
+						<View gridArea="data" UNSAFE_className="big-number-data">
+							{formattedValue}
+						</View>
+						<Flex gap="size-50">
+							{cloneElement(icon, { size: iconSize })}
+							{label}
+						</Flex>
+					</View>
+				) : (
+					<>
+						{icon && (
+							<View gridArea="icon" alignSelf={iconAlign} justifySelf={iconJustify}>
+								{cloneElement(icon, { size: iconSize })}
+							</View>
+						)}
+						<View gridArea="label" alignSelf={labelAlign} UNSAFE_className="big-number-label">
+							{label}
+						</View>
+					</>
+				)}
+				{!displayCombo && (
+					<View gridArea="data" UNSAFE_className="big-number-data">
+						{formattedValue}
+					</View>
+				)}
+			</Grid>
+		</div>
+	);
+};
+
+BigNumber.displayName = 'BigNumber';
+
+function getGridProperties(
+	icon: IconElement | undefined,
+	orientation: Orientation,
+	height: number | undefined,
+	chartWidth: number,
+	lineProps?: LineProps
+) {
+	let labelAlign: 'start' | 'center' = 'start';
+	let iconAlign: 'start';
+	let iconJustify: 'end';
+	const iconSize = getIconSizeByOrientation(
+		orientation,
+		chartWidth,
+		icon !== undefined,
+		lineProps !== undefined,
+		height
+	);
+
+	// First block (line elements)
+	if (lineProps) {
+		// Then get the icon conditions
+		if (icon) {
+			if (orientation === 'vertical') {
+				labelAlign = 'center';
+				iconJustify = 'end';
+				return {
+					areas: ['sparkline', 'combo', 'data'],
+					columns: ['4fr'],
+					displayCombo: true,
+					labelAlign,
+					iconJustify,
+					iconSize,
+				};
+			}
+			if (orientation === 'horizontal') {
+				iconJustify = 'end';
+				iconAlign = 'start';
+				labelAlign = 'center';
+				return {
+					areas: ['sparkline combo'],
+					columns: ['1fr'],
+					iconJustify,
+					iconAlign,
+					displayCombo: true,
+					iconSize,
+					labelAlign,
+				};
+			}
+		} else {
+			if (orientation === 'vertical') {
+				return { areas: ['sparkline', 'data', 'label'], columns: ['1fr'], labelAlign };
+			}
+			if (orientation === 'horizontal') {
+				return { areas: ['sparkline data', 'sparkline label'], columns: ['1fr 1fr'], labelAlign };
+			}
+		}
```

Comment:
I would just extract this into a separate function since it's adding a lot to the complexity of the function.